docs/advanced/extensions.md

```diff
@@ -0,0 +1,63 @@
+---
+icon: material/apps
+---
+
+`HTTPX` provides an extension mechanism to allow additional information 
+to be added to requests and to be returned in responses. `hishel` makes use
+of these extensions to expose some additional cache-related options and metadata.
+These extensions are available from either the `hishel.CacheClient` / 
+`hishel.AsyncCacheClient` or a `httpx.Client` / `httpx.AsyncCacheClient`
+using a `hishel` transport.
+
+## Request extensions
+
+`hishel` currently provides only one extension on requests - `cache_disabled`. This
+extension temporarily disables the cache by passing appropriate RFC911 headers to
+ignore cached responses and to not store incoming responses. For example:
+
+```python
+>>> import hishel
+>>> client = hishel.CacheClient()
+>>> response = client.get("https://www.example.com/cacheable-endpoint", extensions={"cache_disabled": True})
+
+```
+This feature is more fully documented in the [User Guide](/userguide/#temporarily-disabling-the-cache)
+
+## Response extensions
+
+`hishel` provides two extensions on responses that provide additional information regarding whether a response was returned from the cache, and additional metadata about the cached response when a response is returned from the cache. 
```

Comment:
Let's avoid specifying how many extensions we currently support; I believe it will complicate the maintenance process (we will need to change it with each extension).
Instead, we can do something similar to the headers documentation we have, see the [doc](https://hishel.com/advanced/http_headers/) (simple, minimalistic examples and documentation for each of them).

Reply:
Understood, addressed in dc4f878